File: frontend/types/api-key.ts
/**
 * API 密钥配置相关类型定义
 */

// API 服务类型
export type ApiKeyService = 'tmdb' | 'google_books' | 'bangumi';

// 测试状态
export type TestStatus = 'not_tested' | 'success' | 'failed';

// API 密钥配置
export interface ApiKeyConfig {
  service: string;
  api_key?: string | null;  // 完整密钥（仅在 reveal=true 时返回）
  api_key_preview: string | null;
  has_key: boolean;
  base_url: string | null;
  enabled: boolean;
  last_tested_at: string | null;
  test_status: TestStatus;
  test_message: string | null;
  description: string | null;
  created_at: string;
  updated_at: string;
}

// API 密钥配置列表响应
export interface ApiKeyListResponse {
  configs: ApiKeyConfig[];
}

// API 密钥更新数据
export interface ApiKeyUpdateData {
  api_key?: string;
  base_url?: string;
  enabled?: boolean;
  description?: string;
}

// API 连接测试结果
export interface ApiKeyTestResult {
  success: boolean;
  message: string;
  tested_at: string;
}

// 服务信息（用于UI展示）
export interface ServiceInfo {
  service: ApiKeyService;
  name: string;
  description: string;
  icon: string;
  baseUrlRequired: boolean;
  defaultBaseUrl: string;
}

File: backend/app/schemas/api_key.py
"""
API 密钥配置 Schemas
"""
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum


class ApiKeyServiceEnum(str, Enum):
    """API 服务枚举"""
    TMDB = "tmdb"
    GOOGLE_BOOKS = "google_books"
    BANGUMI = "bangumi"


class TestStatusEnum(str, Enum):
    """测试状态枚举"""
    NOT_TESTED = "not_tested"
    SUCCESS = "success"
    FAILED = "failed"


class ApiKeyConfigResponse(BaseModel):
    """API 密钥配置响应模型"""
    service: str
    api_key: Optional[str] = Field(None, description="API 密钥完整值（仅在 reveal=true 时返回）")
    api_key_preview: Optional[str] = Field(None, description="API 密钥预览（脱敏）")
    has_key: bool = Field(description="是否已配置密钥")
    base_url: Optional[str] = Field(None, description="API 基础 URL")
    enabled: bool = Field(description="是否启用")
    last_tested_at: Optional[datetime] = Field(None, description="最后测试时间")
    test_status: str = Field(description="测试状态")
    test_message: Optional[str] = Field(None, description="测试消息")
    description: Optional[str] = Field(None, description="配置描述")
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class ApiKeyConfigUpdate(BaseModel):
    """API 密钥配置更新请求"""
    api_key: Optional[str] = Field(None, description="API 密钥")
    base_url: Optional[str] = Field(None, description="API 基础 URL")
    enabled: Optional[bool] = Field(None, description="是否启用")
    description: Optional[str] = Field(None, description="配置描述")


class ApiKeyTestResponse(BaseModel):
    """API 连接测试响应"""
    success: bool = Field(description="测试是否成功")
    message: str = Field(description="测试消息")
    tested_at: datetime = Field(description="测试时间")


class ApiKeyListResponse(BaseModel):
    """API 密钥配置列表响应"""
    configs: List[ApiKeyConfigResponse]

